src/App.tsx

```diff
@@ -1,7 +1,38 @@
 import React from 'react';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <p className="App__message">The last pressed key is [Enter]</p>
-  </div>
-);
+type State = {
+  pressedKey: string | null;
+};
+
+// eslint-disable-next-line
+export class App extends React.Component<{}, State> {
+  state = {
+    pressedKey: null,
+  };
+
+  componentDidMount() {
+    document.addEventListener('keyup', (event: KeyboardEvent) => {
+      this.setState({ pressedKey: event.key });
+    });
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('keyup', (event: KeyboardEvent) => {
+      this.setState({ pressedKey: event.key });
+    });
+  }
+
```

Comment:
I think, you can move  `(event: KeyboardEvent) => { this.setState({ pressedKey: event.key })`  to a function and use it in `componentDidMount `and in `componentWillUnmount `like this:

```
componentDidMount() {
  document.addEventListener('keyup', this.handleKeyUp);
}

componentWillUnmount() {
  document.removeEventListener('keyup', this.handleKeyUp);
}

handleKeyUp = (event: KeyboardEvent) => {
  this.setState({ pressedKey: event.key });
};
```